libp2p/protocols/rendezvous.nim

```diff
@@ -734,8 +734,8 @@ proc new*(
         await rdv.discover(conn, msg.discover.tryGet())
       of MessageType.DiscoverResponse:
         trace "Got an unexpected Discover Response", response = msg.discoverResponse
-    except CancelledError as exc:
-      raise exc
+    except CancelledError:
+      trace "cancelled rendezvous handler"
     except CatchableError as exc:
```

Comment:
after specifying raises, it's good to check if these broad handlers can be removed - removing broad exception handlers or commenting why they are broad is one way to ensure that the conversion is complete

Reply:
i am doing that, but not at every place. this task has already exceed planned time for it. sometimes going in and taking care of those details, requires more time to understand logic and implement correct behavior. not to mention that sometime behavior is not specified for differed cases nor it is covered with tests.

i will suggest to have another round of refactoring effort where `except CatchableError` and raising `CatchableError` should be specified with specific error or commented with explanation why `CatchableError` is used.

Reply:
> i will suggest to have another round of refactoring

in that case, one thing _not_ to change is which exceptions get caught and which are raised - ie in cases like this, leave the CancelledError to propagate - this should be done before proceeding any further, ie otherwise there is risk to introduc subtle issues.

Reply:
got it! 

i was not thinking about `CancelledError` here (this discussion is in comment above). i wanted to mention that there are places where `CatchableError` is used but it might have been more suitable for `ValueError` (narrowing down exceptions).

Reply:
>  more suitable for ValueError (narrowing down exceptions).


Yeah, as a general approach, we _tend to_ remap random exceptions like this to something appropriate for the abstraction, when working with abstractions like libp2p tends to do - this is above all important in order not to leak implementation details into the "exception ABI", ie the set of exceptions that the consumer of the library is expected to handle - changing raised exceptions to something outside of the given raises hierachy is an ABI break which (in theory) requires a new major release.